### DISCUSSION TITLE
Position class is getting applied only at certain values (50)

### DISCUSSION THREAD

**Discussion by moyscode** [tailwindlabs/tailwindcss#12828] [Help]
**What version of Tailwind CSS are you using?**
v3.4.1 

**What build tool (or framework if it abstracts the build tool) are you using?**
SvelteKit with Svelte 5.0.0-next.1,
I have followed the guide https://tailwindcss.com/docs/guides/sveltekit for installing & using TailwindCSS

**What version of Node.js are you using?**
v18.17.1

**What browser are you using?**
Firefox & Chrome

**What operating system are you using?**
Windows 10

**Reproduction URL**
Unfortunately I am unable to find a playground where I can put my code.
I even tried this URL https://svelte.dev/repl/f2a988ddbd5644f18d7cd4a4a8277993?version=4.2.9.
I use `import { browser } from '$app/environment';` in my code & get the error doe "$app" really exists!

I am putting my whole code here (there is nothing confidential in there).
```
<script lang="ts">
	import { browser } from '$app/environment';

	let value = 50;

	if (browser) {
		const feClassList = document!.querySelector('.fe')!.classList;
		// feClassList.add('block');
		// feClassList.remove('hidden');

		const fseClassList = document!.querySelector('.fse')!.classList;
		fseClassList.add('opacity-0');

		const changePosition = (value: any) => {
			var diff = Array.from(beClassList).filter((element) => element.startsWith('left-'));
			diff.forEach((item) => {
				beClassList.remove(item);
			});
			beClassList.add(`left-[${value}]`);
		};

		const slider = document!.querySelector('#range');
		const beClassList = document!.querySelector('.be')!.classList;
		beClassList.add('block');
		beClassList.remove('hidden');

		slider!.addEventListener('input', (e) => {
			let sliderValue = value + '%';
			changePosition(sliderValue);

			// var diff = Array.from(beClassList).filter((element) => element.startsWith('left-'));
			// diff.forEach((item) => {
			// 	beClassList.remove(item);
			// });
			// beClassList.add(`left-[${sliderValue}]`);
		});
	}
</script>

<main class="grid isolate">
	<section class="grid grid-cols-1 md:grid-cols-2 m-5 gap-1">
		<div class="">
			<p>Hi,</p>
			<div class="[&_h1]:text-4xl [&_h1]:font-black isolate overflow-x-clip">
				<h1 class="">Welcome Mario</h1>
				<div class="relative">
					<h1 class="absolute w-full z-10 fe hidden">Plumber</h1>
					<h1 class="absolute w-full z-10 top-0 right-0 bottom-0 be">Mechanic</h1>
					<input
						type="range"
						id="range"
						class="absolute w-full z-10"
						min="0"
						max="100"
						step="1"
						bind:value
					/>
					<h1 class="fse z-20">Engineer</h1>
				</div>
			</div>
			<p>I repair things.</p>
		</div>
		<div></div>
	</section>
</main>

```
**Describe your issue**
The `left` position (with a arbitrary value) is working only with the vaule 50!
I want to be able to position my element (in this case h1 element with class "be" which has the text "Mechanic"), 
The class gets applied to the position but the position does not change except for when the class is `left-[50%]`. I do not understand why it works only as the value 50% and not with other values. The other values can be see in the Dev tools.

Out of curiosity I changed from `%` to `px` and the behavior is same! When I value is `50px` the `left` is applied, otherwise not.

The expected behavior is for the `left` position (with a arbitrary value) to be applied at all times not only with value 50.
Few pics below showing the issue

Please have a look. This is my first Svelte & Tailwind project, so if I am doing something wrong, please let me know.

Page when it loads
![image](https://github.com/tailwindlabs/tailwindcss/assets/92194668/44ed3ae9-9c11-4734-8c2b-f51b6f0399b7)

Page when Slider is at 50%
![image](https://github.com/tailwindlabs/tailwindcss/assets/92194668/7ae50262-f6ae-4ec6-89aa-ed2f0026918d)

Page when Slider is at a value other than 50%
![image](https://github.com/tailwindlabs/tailwindcss/assets/92194668/88453840-ec17-41d0-8079-c4afbaf917c1)

**Comment by adamwathan** [comment]
Hey! See the relevant documentation on dynamic class names here:

https://tailwindcss.com/docs/content-configuration#dynamic-class-names

Tailwind doesn't run your JS when compiling your CSS, it only ever sees `left-[${value}]`, not the final string that the browser sees. You'll want to use inline styles for this sort of situation 👍

**Reply by moyscode** [reply]
@adamwathan Thanks a lot of the clarification. Your guidance helped. I used inline styles as you suggested & it works. Thanks.
